src/java.base/share/classes/java/security/PEMRecord.java

```diff
@@ -82,7 +84,9 @@ public record PEMRecord(String type, String pem, byte[] leadingData)
      * @throws IllegalArgumentException on incorrect input values.
      */
     public PEMRecord(String type, String pem, byte[] leadingData) {
-        if (type == null && pem == null && leadingData == null) {
+        this.leadingData = (leadingData == null ? null : leadingData.clone());
```

Comment:
Not sure we should do the cloning at creation and in the getter. A record is _known_ to be only shallowly immutable and users should be prepared for this, for example, do not pass it to an untrusted method if they want to reuse it. I'd rather change this back to a normal class if you believe array cloning is necessary.